core/src/builder/parachain.rs

```diff
@@ -273,11 +273,35 @@ where
 				.ok()
 				.flatten()?;
 
+			let upward_messages = collation_info
+				.upward_messages
+				.try_into()
+				.map_err(|e| {
+					tracing::error!(
+						target: DEFAULT_COLLATOR_LOG_TARGET,
+						error = ?e,
+						"Could not cast upward messages.",
+					)
+				})
+				.ok()?;
```

Comment:
I'm asking here, but the question goes more for all these patterns. If `fudge` intention is to be used for testing. Why, instead of tracing and propagating a None, do we no just unwrap everywhere? That way, when it doesn't work, the error just panic with the correct message in the correct line which generate it.

Reply:
Maybe @mustermeiszer can provide a better answer to this ^^. 

In my fudge branch for xcm tests I'm addressing some of these and propagating some more errors in cases similar to this.

Reply:
> unwrapping in a dependency library is not really an option IMO.

Ok, this makes sense 👍🏻